File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/mount/PaddockSellRequestMessage.scala





















// Generated on 05/03/2013 10:34:42
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PaddockSellRequestMessage extends MessageDeserializer {
  val messageId = 5953

  protected trait Message extends BaseMessage with Serializable {
    val messageId = PaddockSellRequestMessage.messageId

    val price: Int

    def serialize(buf: Buffer) {
      buf.writeInt(price)
    }

  }

  protected def create(price: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val price = buf.readInt()

    create(price)
  }

}

object PaddockSellRequestMessage extends PaddockSellRequestMessage {
  case class MessageImpl(price: Int) extends Message
  type Target = MessageImpl

  override def create(price: Int) = MessageImpl(price)

  def apply(price: Int) = create(price)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PaddockSellRequestMessage.java


// Generated on 05/08/2013 19:37:45
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PaddockSellRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5953;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int price;
    
    public PaddockSellRequestMessage() { }
    
    public PaddockSellRequestMessage(int price) {
        this.price = price;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(price);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        price = buf.readInt();
        if (price < 0)
            throw new RuntimeException("Forbidden value on price = " + price + ", it doesn't respect the following condition : price < 0");
    }
    
}
